src/corppass/corppass-helper-ndi.ts

```diff
@@ -165,11 +165,18 @@ export class NdiOidcHelper {
 			const {
 				data: { keys },
 			} = await this.axiosClient.get<{ keys: Object[] }>(finalJwksUri, { headers: this.additionalHeaders });
-			const jwsVerifyKey = JSON.stringify(keys[0]);
 
 			const { access_token } = tokens;
-			const verifiedJws = await JweUtil.verifyJWS(access_token, jwsVerifyKey, "json");
-			return JSON.parse(verifiedJws.payload.toString()) as AccessTokenPayload;
+			let error = null;
+			for (const key of keys) {
+				try {
+					const verified = await JweUtil.verifyJWS(access_token, JSON.stringify(key), "json");
+					return JSON.parse(verified.payload.toString()) as AccessTokenPayload;
+				} catch (err) {
+					error = err;
+				}
+			}
+			throw error;
```

Comment:
Is it safe to simply `throw error;`?
Or should we be doing a check on whether there's an error before throwing?

Reply:
ideally if it reaches there it means all of the verification failed and there should be at least one error; of course to be extra safe it can be

```js
if (error) {
  throw error;
} else {
  throw new Error('all keys failed');
}
```